lib/deaf_grandma.rb

```diff
@@ -20,12 +20,34 @@ def run!
 
   def speak(input)
 
-    #Implement your code here <<<<<<<<<
-
+    if is_yelling(input)
+        "NOT SINCE 1964!"
+    else
+        "SPEAK UP SONNY!"
+    end
   end
 
   private
 
+    def is_yelling(text)
+        return false if text != text.upcase
+
+        if text == "BYE"
+            @bye_counter += 1
```

Comment:
So, how about doing something like this then?

```ruby
def check_goodbye(text)
  if text == "BYE"
    @bye_counter += 1
  end

  if @bye_counter == 3
    say_goodbye
  end
end

def is_yelling?(text)
  return text == text.upcase
end

def speak(input)
  if is_yelling?(input)
    check_goodbye(input)

    "NOT SINCE 1964!"
  else
    "SPEAK UP SONNY!"
  end
end
```